core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -2715,6 +2715,66 @@ private[spark] object Utils extends Logging {
     HashCodes.fromBytes(secretBytes).toString()
   }
 
+  /**
+   * A safer version than scala obj's getClass.getSimpleName and Utils.getFormattedClassName
+   * which may throw Malformed class name error.
+   * This method mimicks scalatest's getSimpleNameOfAnObjectsClass.
+   */
+  def getSimpleName(fullyQualifiedName: String): String = {
```

Comment:
How about changing the method signature into `def getSimpleName(obj: AnyRef)`?
Then, we simply handle two cases; one is correct and the other non-correct?, e.g.,
```
def getSimpleName(obj: AnyRef): String = {
  if (incorrect case) {
    // Canonicalizes the name for correction
    ...
  } else {
    // If no problem, just returns getSimplename
    obj.getClass.getSimpleName
  }
}
```

Reply:
a if statement to determine if the name is good or not may not be stable and comprehensive - how about we use try-catch here and only use the alternative if obj.getClass.getSimpleName throws the Malformed class name error

Reply:
updated. since getSimpleName is a method from Class, it makes sense for Utils.getSimpleName to also takes a Class param. Also, Utils.getSimpleName will try to use the Class's getSimpleName first, and only use the alternative if the internal error is caught. This should minimize the potential impact of this patch